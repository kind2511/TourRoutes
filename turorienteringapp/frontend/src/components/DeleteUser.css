.delete-user-container {
    width: 500px;
    padding: 20px;
    border-radius: 10px;
    box-shadow: 0px 0px 15px rgba(19, 182, 38, 0.937);
    background-color: #d5d2d2;
    position: fixed;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    display: flex;
    flex-direction: column;
    align-items: center;
}

.delete-user-container h1 {
    margin-bottom: 20px;
    font-size: 24px;
}

.delete-user-container p {
    margin-bottom: 20px;
    font-size: 16px;
    font-weight: bold;
    padding-left: -100px;
    
}

.button-group {
    display: flex;
    justify-content: space-between;
    width: 100%;
}

.delete-user-container .delete-button,
.delete-user-container .cancel-button {
    padding: 10px 20px;
    margin: 0 5px;
    border: none;
    border-radius: 5px;
    cursor: pointer;
    font-size: 16px;
    color: white;
}

.delete-user-container .delete-button {
    background-color: #E53E3E;
}

.delete-user-container .cancel-button {
    background-color: #4CAF50;
}

.delete-user-container button:hover {

    background-color: #61dafb;
    color: #333;
    
}
