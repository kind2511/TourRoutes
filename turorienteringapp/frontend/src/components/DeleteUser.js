import React from 'react';
import { useNavigate } from 'react-router-dom';
import './DeleteUser.css';

const DeleteUser = () => {
  const navigate = useNavigate();

  /**
 * Handles the account deletion process.
 * Sends a DELETE request to the server to remove the user's profile.
 * On success, logs a message and redirects to the login page.
 * Catches and logs any errors that occur during the process.
 */

  const handleDelete = async () => {
    try {
      const response = await fetch('http://localhost:8000/api/v1/users/deleteMyProfile', {
        method: 'DELETE',
        headers: {
          'Content-Type': 'application/json',
          Authorization: 'Bearer ' + localStorage.getItem('token'),
        },
        credentials: 'same-origin',
      });

      if (!response.ok) {
        throw new Error(`HTTP error! status: ${response.status}`);
      } else {
        console.log('Account deleted successfully');
        navigate('/login'); // Redirect to login after account deletion
      }
    } catch (error) {
      console.error('Error deleting account:', error);
    }
  };

  return (
    <div className="delete-user-container">
      <div className="delete-confirmation-box">
        <p>Are you sure you want to delete your account?</p>
        <p>This action cannot be undone.</p>
        <div className="delete-buttons">
          <button className="delete-account-btn" onClick={handleDelete}>Delete Account</button>
          <button className="cancel-btn" onClick={() => navigate(-1)}>Cancel</button>
        </div>
      </div>
    </div>
  );
};

export default DeleteUser;
