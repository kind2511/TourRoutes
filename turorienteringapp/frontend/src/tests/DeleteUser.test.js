import React from 'react';
import { render } from '@testing-library/react';
import DeleteUser from '../components/DeleteUser';
import { BrowserRouter } from 'react-router-dom';

test('renders DeleteUser component without crashing', () => {
  render(
    <BrowserRouter>
      <DeleteUser />
    </BrowserRouter>
  );
});

//------------------------------------------------------>

test('DeleteUser component snapshot', () => {
  const { asFragment } = render(
    <BrowserRouter>
      <DeleteUser />
    </BrowserRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});



