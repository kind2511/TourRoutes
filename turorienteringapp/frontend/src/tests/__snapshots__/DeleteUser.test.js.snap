// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteUser component snapshot 1`] = `
<DocumentFragment>
  <div
    class="delete-user-container"
  >
    <h1>
      Delete Account
    </h1>
    <p>
      Are you sure you want to delete your account?
    </p>
    <p>
      This action cannot be undone.
    </p>
    <div
      class="delete-buttons"
    >
      <button
        class="delete-account-btn"
      >
        Delete Account
      </button>
      <button
        class="cancel-btn"
      >
        Cancel
      </button>
    </div>
  </div>
</DocumentFragment>
`;
